src/Components/Server/src/Circuits/DefaultCircuitFactory.cs

```diff
@@ -64,15 +77,45 @@ public override CircuitHost CreateCircuitHost(HttpContext httpContext, IClientPr
                 client,
                 rendererRegistry,
                 renderer,
-                config,
+                components,
+                dispatcher,
                 jsRuntime,
                 circuitHandlers);
 
-            // Initialize per-circuit data that services need
-            (circuitHost.Services.GetRequiredService<IJSRuntimeAccessor>() as DefaultJSRuntimeAccessor).JSRuntime = jsRuntime;
+            // Initialize per - circuit data that services need
             (circuitHost.Services.GetRequiredService<ICircuitAccessor>() as DefaultCircuitAccessor).Circuit = circuitHost.Circuit;
 
             return circuitHost;
         }
+
+        private static IList<ComponentDescriptor> ResolveComponentMetadata(HttpContext httpContext, IClientProxy client)
+        {
+            if (client != null)
+            {
+                var endpointFeature = httpContext.Features.Get<IEndpointFeature>();
+                var endpoint = endpointFeature?.Endpoint;
+                if (endpoint == null)
+                {
+                    throw new InvalidOperationException(
+                        "ComponentHub doesn't have an associated endpoint. " +
```

Comment:
`$"{nameof(ComponentHub)} does not have an associated endpoint." Use 'app.UseRouting(routes => routes.MapComponentHub<App>(\"app\"))' to register your hub in the application startup code.`